### DISCUSSION TITLE
Plans for a replacement of <ModalViewWrapper />

### DISCUSSION THREAD

**Discussion by andioneto** [twilio-labs/paste#639] [Q&A] [answered]
Hi folks 👋 I'm wondering if there is a replacement utility for the [<ModalViewWrapper />](https://code.hq.twilio.com/react/react-components/blob/master/packages/ui-component-modal/src/js/ModalViewWrapper.js) in Paste? If it does not exist, are the plans to add it to the component library?

If not, then is that an experience we are trying to fade out of? It occurs in a few places, but I am most familiar with it in the context of associating an emergency validated address to a phone number on the number instance page. Essentially, after the form containing the address information is submitted, if the address is not in a valid format (according to providers), the view switches to a "suggested address."

**Comment by SiTaggart** [comment]
Hi @andipants12 I'm not familiar with `ModalViewWrapper` and I can't seem to find it in the react-components docs. I'm also not sure many on the team, most of which are new to Console relatively speaking, would be familiar either.

Do you mind going into further detail on the experience? Whats the flow? Maybe a screenshot  or 2?

**Reply by corinne-nagel** [reply]
Hey @andipants12! Did you find a solution for this?

**Comment by andioneto** [comment]
Hi @SiTaggart and @corinne-nagel, please excuse the delayed response!

I'm attaching a screen capture of the old component docs with the animated multi views.
![modalanimated](https://user-images.githubusercontent.com/18120906/90183864-157c6580-dd69-11ea-9dbf-233485f4e157.gif)

**Reply by andioneto** [reply]
So no, still have not made a resolution, but I was speaking to @vnguyen94 about this and we were thinking of making a simple stateful wrapper on our own with animation. Wasn't sure though if that would be advised.

**Reply by vankhoawin** [reply]
we have the building blocks:

https://github.com/twilio-labs/paste/blob/master/packages/paste-core/components/modal/src/ModalBody.tsx (with some conditional rendering based on view)
https://github.com/twilio-labs/paste/tree/master/packages/paste-libraries/animation (for animation between views)

**Reply by andioneto** [reply]
Great, thanks Van. I suppose the ask/question here then is if this is in the roadmap (didn't see it, but it's likely under a name I didn't recognize) and if there has been any early stage planning? Would like to ensure that we create a temp component wrapper with the long term plans in mind to ease migration down the road.

**Comment by SiTaggart** [comment, marked as answer]
Hi Folks

So in the Modal design process the concept of a stepped modal was cut pretty early on in terms of scope. We just didn't have the time to think about or design the best, most generic stepping modal experience that would suit everyones need. 

What we leaned on was composition. We give you all the pieces you need to build you're own stepped modal experience. A footer with aligned actions, and a body for any content you desire. If you wish to provide animation to that too, as @vnguyen94 points out you can use the animation library.

But I think we'd strongly suggest that it is still lead by the deign partner in your BU though so they can come through Office hours, crits and UX reviews to design the best experience.

There is still a possibility we might do somehting in the future, but it's not on the roadmap at all right now, so it's unlikely.

**Reply by andioneto** [reply]
Ok great! AFAIK numbers console does not currently have a dedicated designer, VNV Console has three folks, but I know their bandwidth is pretty limited. I think in the short term, @vnguyen94 and I can work together to create a simple wrapper and we can present the POC to the designers on my team as a compromise. Thank you!
